src/_providers.tf

```diff
@@ -13,7 +13,11 @@ terraform {
 }
 
 provider "azurerm" {
-  features {}
+  features {
```

Comment:
As far as I know, it's not dangerous at all. Nothing in the [resource group](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group) forces a destroy/recreate if something is changed. 

The real problem is that the `terraform destroy` command on an Azure resource group will check to see if there are any lingering resources in the resource group. If there are, it'll cancel the operation.
```
[prevent_deletion_if_contains_resources](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/features-block#prevent_deletion_if_contains_resources) - (Optional) Should the azurerm_resource_group resource check that there are no Resources within the Resource Group during deletion? This means that all Resources within the Resource Group must be deleted prior to deleting the Resource Group. Defaults to true.
```
Sometimes during an intentional `terraform destroy`, there may be resources left in the resource group that get created but aren't referenced in the terraform to be destroyed. If that's the case, the resources remain in the resource group. Using the Azure portal, deleting a resource group automatically deletes everything inside that resource group, no questions asked. This feature mimics that functionality.

I will add onto this that I've faced countless errors both locally and in the MD UI where I've tried to change or delete a resource, and I'm faced with an error that says the operation was cancelled because resources remain in the resource group, and the terraform error explicitly asks me to add this feature into the provider. Sadly, it's becoming a standard requirement at this point.

Reply:
My fear is this.

- create a resource group
- put a database in it
- delete the resource group

The database is nuked as well. That's the "nice" part about RGs yeah? That's what feels potentially destructive here. 

Since you know Azure the best, I approved this. However, I'd love to open tickets in our backlog and capture these errors to understand them better. If it's something we have to add to "make Azure work", then sparingly but I'd like to be cautious about it.

Reply:
If someone is deleting the resource group, they're doing it full-well knowing that everything inside will get destroyed as well. Resources in Azure cannot exist w/o a resource group. Databases don't need resource groups, but they need servers. So, if the server the database is attached to is intentionally deleted, well so is the database. If the customer wants to keep their database after tearing down infra, they need to back it up.